components/common/GetReady.tsx

```diff
@@ -0,0 +1,190 @@
+  const mainControls = useAnimation();
+  const isInView = useInView(getReadyRef, { once: false });
+
+  const handleSubscribe = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    const token = executeRecaptcha && (await executeRecaptcha('newsletter_subscribe'));
+    if (!token) {
+      toast.error("We've detected a bot-like behavior. Please try again later.");
+      return;
+    }
+
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+
+    if (email.length === 0 || !emailRegex.test(email)) {
+      toast.error('Please enter a valid email address!');
+      return;
+    }
+
+    setLoading(true);
+
+    try {
+      const response = await axios.post(
+        '/api/subscribe',
+        {
+          email,
+          attributes: {
+            IS_PROVIDER: isProvider ? 'Yes' : 'No',
+            OPT_IN: true,
+          },
+        },
+        {
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        },
+      );
+      if (response.status === 200) {
+        toast.success("You've been subscribed!");
+        setEmail('');
+      }
+    } catch (error: any) {
+      toast.error('An error occurred. Please try again!');
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    if (isInView) {
+      mainControls.start('visible');
+    } else {
+      mainControls.start('hidden');
+    }
+  }, [isInView, mainControls]);
+
+  return (
+    <div
+      className="relative mt-[4rem] flex min-h-[55vh] w-full flex-col items-center justify-center gap-8 sm:mt-0 md:mt-20 md:min-h-[50vh] lg:h-screen"
+      id="join"
+    >
+      <Toaster position="top-right" toastOptions={{ duration: 5000 }} reverseOrder={false} />
+      <Image
+        src={backgroundImage}
+        alt={waveImageAlt}
+        className="absolute left-0 top-[20%] w-[110%] sm:top-[0%]"
+        width={1376}
+        height={396}
+      />
+      <div>
+        <WordAnimation variant="desktop" content={content} />
+        <WordAnimation variant="tablet" content={content} />
+        <WordAnimation variant="mobile" content={content} />
+      </div>
+
+      <motion.form
+        ref={getReadyRef}
+        variants={{
+          hidden: { opacity: 0 },
+          visible: { opacity: 1 },
+        }}
+        initial="hidden"
+        animate={mainControls}
+        onSubmit={handleSubscribe}
+        transition={{ duration: 0.3, delay: 0.6, ease: 'easeIn' }}
+        className="ps-slim border-[#111928] mt-[1rem] flex h-[56px] w-[95%] items-center justify-between rounded-[2rem] border-2 py-1 pl-4 pr-1 sm:mt-0 md:w-[549px] lg:w-[458px]"
+      >
+        <input
+          type="email"
+          name="newsletter"
+          id="newsletter"
+          required
+          placeholder="Enter your email address."
+          className="h-full w-[90%] border-none outline-none"
+          value={email}
+          onChange={(e) => setEmail(e.target.value)}
+        />
+        <button
```

Comment:
Could we turn this button into its own component as well? We have a few buttons that look similar.
<img width="279" alt="Screenshot 2024-12-03 at 1 06 59 AM" src="https://github.com/user-attachments/assets/c858412d-e211-4e13-9e68-c08dc5d5cd6c">
<img width="96" alt="Screenshot 2024-12-03 at 1 07 15 AM" src="https://github.com/user-attachments/assets/888676b2-1fc3-45b9-b509-1966f427843b">

Reply:
I created the Submit component inside 'components/common'. 'Be a Genie' and 'Join Now' are already separate components. BeAGenie.tsx and JoinNowButton.tsx inside common.
Please let me know if i understand it correctly.

Reply:
thank you @Muhammad-Umar-Khan but what i meant was the make a 'Button.tsx' component that can be used for all these buttons above.

- Have a prop for `text`, so parent can pass in `Submit` or `Be a Genie` or `Join Now`
- Allow the background color and border to be customizable
- And like you have it now the `loading` prop